File: solution/eightqueens.js
const check = (index, solution)=>{
    for (let i = 0; i< index; i++){
        if (solution[i] == solution[index] || solution[i] + (index - i) == solution[index] || solution[i] - (index - i) == solution[index] ){
            return false
        }
    }
    return true
}

const eightqueens = (index, solution)=>{
    if (index == 8){
        console.log(solution.join(''))
    } else {
        for (let i = 1; i < 9; i++){
            solution[index] = i
            if (check(index,solution)){
                eightqueens(index+1, solution)
            }
        }
        eightqueens[index] = 0
    }
}

let solution = new Array(8).fill(0)
let index = 0
eightqueens(index, solution)

File: solution/eightqueens.py
def check(index, solution):
    for i in range(0,index):
        if solution[i] == solution[index] or solution[i] + (index - i) == solution[index] or solution[i] - (index - i) == solution[index]:
            return False
    return True

def eightqueens(index, solution):
    if index == 8:
        for n in solution:
            print(n,end = '')
        print()
    else:
        for i in range(1,9):
            solution[index] = i
            if check(index, solution):
                eightqueens(index+1,solution)
        solution[index]= 0

if __name__ == '__main__':
    solution = [0]*8
    index = 0
    eightqueens(index, solution)

